src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -716,6 +778,73 @@ discover_dfuse_mounts(void)
 	return rc;
 }
 
+static int
+fetch_dfs_cont_file_obj_with_fd(int fd, struct dfs_mt *dfs_mt, dfs_obj_t **obj)
+{
+	int                    cmd, rc;
+	d_iov_t                iov	 = {};
+	char                  *buff_obj = NULL;
+	struct dfuse_hsd_reply hsd_reply;
+	struct dfuse_il_reply  il_reply;
+
+	rc = ioctl(fd, DFUSE_IOCTL_IL, &il_reply);
+	if (rc != 0) {
+		rc = errno;
+		if (rc != ENOTTY)
+			DS_WARN(rc, "ioctl call on %d failed", fd);
```

Comment:
Even fetch_dfs_cont_file_obj_with_fd() failed, open() still can return the fd allocated by kernel. If you think error is more appropriate here, I can change it.

Reply:
This was more of a question to you..
Warning messages should be used when something that the user is expecting but another path was taken and the operation is still onging / succeeds.
Error messages are for when we get legitimate errors from DAOS that something went wrong like EIO, etc.
otherwise, like if user is opening a non-existent file, this is more of a debug message that can be logged at the DAOS log level.

Reply:
ok. I will change this and several other warnings in this function to debug message. Thank you!

Reply:
DS_WARN does print errno and string version message.
fetch_dfs_cont_file_obj_with_fd() is called inside open. When it is called, we already get fd from dfuse without issue. Even fetch_dfs_cont_file_obj_with_fd fails, we still return fd allocated by dfuse. I think warning is appropriate here.